Dart/thirdPartySrc/analysisServer/org/dartlang/analysis/server/protocol/AnalysisError.java

```diff
@@ -119,12 +119,12 @@ public boolean equals(Object obj) {
   public static AnalysisError fromJson(JsonObject jsonObject) {
     String severity = jsonObject.get("severity").getAsString();
     String type = jsonObject.get("type").getAsString();
-    String code = jsonObject.get("code") == null ? null : jsonObject.get("code").getAsString();
     Location location = Location.fromJson(jsonObject.get("location").getAsJsonObject());
     String message = jsonObject.get("message").getAsString();
     String correction = jsonObject.get("correction") == null ? null : jsonObject.get("correction").getAsString();
+    String code = jsonObject.get("code").getAsString();
```

Comment:
@devoncarew This makes Dart plugin incompatible with SDKs older than 1.16 because of an NPE. I guess there's and error in spec that is used to generate this code and 'code' attribute should be marked as optional there. 

// cc @pq (related commit: https://github.com/dart-lang/sdk/commit/e7e205f00b24faf9ee5cdf0baf21b8a0a04f8c3a)